beluga/include/beluga/random/multivariate_uniform_distribution.hpp

```diff
@@ -0,0 +1,193 @@
+// Copyright 2024 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BELUGA_RANDOM_MULTIVARIATE_UNIFORM_DISTRIBUTION_HPP
+#define BELUGA_RANDOM_MULTIVARIATE_UNIFORM_DISTRIBUTION_HPP
+
+#include <random>
+
+#include <sophus/se2.hpp>
+#include <sophus/se3.hpp>
+
+#include <beluga/sensor/data/occupancy_grid.hpp>
+
+/**
+ * \file
+ * \brief Implementation of a multivariate uniform distribution for SE(2) and SE(3) spaces.
+ */
+
+namespace beluga {
+
+/// Primary template for a multivariate uniform distribution.
+/**
+ * \tparam T The result type for sampling from the distribution.
+ * \tparam Constraint The constraint type to limit the range of the distribution function.
+ */
+template <class T, class Constraint>
+class MultivariateUniformDistribution;
+
+/// Struct representing an axis-aligned bounding region in 2D space.
+struct BoundingRegion2d {
+  double x_min;  ///< Minimum x coordinate of the map.
+  double x_max;  ///< Maximum x coordinate of the map.
+  double y_min;  ///< Minimum y coordinate of the map.
+  double y_max;  ///< Maximum y coordinate of the map.
+};
```

Comment:
@nahueespinosa meta: should we be using [`Eigen::AlignedBox`](https://eigen.tuxfamily.org/dox/classEigen_1_1AlignedBox.html) instead? Same for 3D bounding boxes.

Reply:
Of course! Done in https://github.com/Ekumen-OS/beluga/pull/322/commits/56abf62e4755af5bec6d5aa192bbec25f024b1c2.